gtsam/inference/EliminateableFactorGraph-inst.h

```diff
@@ -236,29 +236,53 @@ namespace gtsam {
     } else {
       // No ordering was provided for the marginalized variables, so order them using constrained
       // COLAMD.
-      bool unmarginalizedAreOrdered = (boost::get<const Ordering&>(&variables) != 0);
-      const KeyVector* variablesOrOrdering =
-        unmarginalizedAreOrdered ?
-        boost::get<const Ordering&>(&variables) : boost::get<const KeyVector&>(&variables);
+      constexpr bool forceOrder = true;
+      Ordering totalOrdering =
+        Ordering::ColamdConstrainedLast((*variableIndex).get(), variables, forceOrder);
+
+      // Split up ordering
+      const size_t nVars = variables.size();
+      Ordering marginalizationOrdering(totalOrdering.begin(), totalOrdering.end() - nVars);
+      Ordering marginalVarsOrdering(totalOrdering.end() - nVars, totalOrdering.end());
+
+      // Call this function again with the computed orderings
+      return marginalMultifrontalBayesNet(std::cref(marginalVarsOrdering), marginalizationOrdering, function, variableIndex);
+    }
+  }
 
+  /* ************************************************************************* */
+  template<class FACTORGRAPH>
+  std::shared_ptr<typename EliminateableFactorGraph<FACTORGRAPH>::BayesNetType>
+    EliminateableFactorGraph<FACTORGRAPH>::marginalMultifrontalBayesNet(
+    const KeyVector& variables,
+    const Eliminate& function, OptionalVariableIndex variableIndex) const
```

Comment:
Could this function get an optional/default argument (or template argument) for `forceOrder` and then the `Ordering` version calls this with `forceOrder=true`?

I guess maybe you didn't want to make the public interface confusing

Reply:
Mostly I want to stay backwards compatible. I thought about the option you suggested, in an extra variant. But that should be part of a de-duplication PR....